update VPN product pages with extension related changes (fix #15983)

wen-2018 opened the pull request:
_If this changeset needs to go into the FXC codebase, please add the `WMO and FXC` label._


## One-line summary

This PR updates the VPN product pages with extension related changes

## Significant changes and points to review

See GH issue for details

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/15983

## Testing

http://localhost:8000/en-US/products/vpn/
http://localhost:8000/en-US/products/vpn/features/
http://localhost:8000/en-US/products/vpn/download/

wen-2018 commented on bedrock/products/templates/products/vpn/landing-refresh.html:
```diff
@@ -177,8 +177,8 @@ <h3 class="c-reason-title">{{ ftl('vpn-landing-blocks-advertisers-from') }}</h3>
       <li>
-        <h3 class="c-reason-title">{{ ftl('vpn-landing-helps-you-access-global') }}</h3>
-        <p>{{ ftl('vpn-landing-check-out-streaming-media') }}</p>
+        <h3 class="c-reason-title">{{ ftl('vpn-landing-take-control') }}</h3>
```
I used new strings here and removed the old ones, instead of using the old ones as fallbacks since they are completely different content

alexgibson replied:
Yeah, I think this makes sense since the copy is directly related to the new image

wen-2018 commented on bedrock/products/templates/products/vpn/download.html:
```diff
@@ -149,15 +149,20 @@ <h3>{{ ftl('vpn-download-also-available') }}</h3>
             </div>
-            <a href="{{ url('products.vpn.windows-download') }}" data-cta-text="VPN Download (Windows)" class="platform-download-link" data-testid="vpn-download-link-secondary-windows">
-              <span class="">{{ ftl('vpn-download-for-windows-long') }}</span>
+            <a class="extension-link" href="#?utm_medium=mozilla-websites&utm_source=mozilla.org&utm_campaign=vpn-download" data-cta-text="Install Firefox Extension" data-testid="vpn-download-link-primary-windows">
```
Waiting for clarification on the link

wen-2018 commented on bedrock/products/templates/products/vpn/download.html:
```diff
@@ -149,15 +149,20 @@ <h3>{{ ftl('vpn-download-also-available') }}</h3>
             </div>
-            <a href="{{ url('products.vpn.windows-download') }}" data-cta-text="VPN Download (Windows)" class="platform-download-link" data-testid="vpn-download-link-secondary-windows">
-              <span class="">{{ ftl('vpn-download-for-windows-long') }}</span>
+            <a class="extension-link" href="https://addons.mozilla.org/firefox/addon/mozilla-vpn-extension/?utm_medium=mozilla-websites&utm_source=mozilla.org&utm_campaign=vpn-download" data-cta-text="Install Firefox Extension" data-testid="vpn-download-link-primary-windows">
```
Still waiting for link confirmation

alexgibson replied:
**Issue: (blocking)** remove `data-testid="vpn-download-link-primary-windows"` as this is going to confuse our functional tests.

alexgibson commented:
I haven’t pulled this PR locally yet to take a look, but I notice there’s quite a lot of new strings here. Do we need to think about fallbacks for older translations at all? We generally try not to show a lot of mixed English copy in places, unless it’s unavoidable. Maybe that discussion has already been had, just calling it out in case.

wen-2018 commented:
> Do we need to think about fallbacks for older translations at all? 

@alexgibson The new strings are on three pages:

- the landing page
 The new strings on this page is corresponding to the new image update. I removed the old ones since they are describing a feature for the old image
 
 - the feature page
 This new string are for a new feature section and the old section with old strings are removed from the page. Since the new section is also added at the top, GH comparison made it look at a lot more strings are changed. In reality, it is removing the last section and adding a new one at the top
 
 - the download page
 This purely adding a new link with a new string

Please let me know your thoughts on if any of the old strings should be used for fallbacks, especially the ones on the landing page

alexgibson commented on bedrock/products/templates/products/vpn/download.html:
```diff
@@ -149,15 +149,20 @@ <h3>{{ ftl('vpn-download-also-available') }}</h3>
-            <div class="platform-body">
-              <h2>{{ ftl('vpn-download-for-windows-v2') }}</h2>
-              <p>{{ ftl('vpn-download-for-windows-requirements') }}</p>
+            <div class="secondary-platform-wrapper">
```
**Issue: (possibly blocking)** I'm not 100% sure what the expected behavior is here, but when visiting on a Windows device, I don't seem to see this new link:

![image](https://github.com/user-attachments/assets/9e24469c-9a47-4a16-b286-dc89f1dd0f4b)

When visiting on macOS (and Windows is a secondary CTA), I can see it though:

![image](https://github.com/user-attachments/assets/3fad726d-677f-4d35-b552-337326a5be55)

wen-2018 replied:
I have clarified with Elise to use the new design showing to Windows users only: https://www.figma.com/design/3Sg25N3kx6oSXHtYA2FXN3/VPN-Updates-Feb-2025?node-id=81-106&m=dev

alexgibson commented on bedrock/products/templates/products/vpn/features.html:
```diff
@@ -205,13 +205,13 @@ <h2 class="c-section-title">{{ ftl('vpn-features-secure') }}</h2>
   <h2 class="c-section-title">{{ ftl('vpn-features-flexible') }}</h2>
 
   {% call split(
```
**Suggestion (non-blocking):** suggest only showing this new feature block for languages that have it translated? 

e.g. `{% if ftl_has_messages('vpn-features-webste-specific-vpn', 'vpn-features-with-the-mozilla-vpn-extention') %}`

alexgibson commented on bedrock/products/templates/products/vpn/landing-refresh.html:
```diff
@@ -290,6 +290,40 @@ <h2 class="c-section-title t-large">{{ ftl('vpn-landing-features') }}</h2>
         </p>
       {% endcall %}
+
+      {% call picto(
```
**Suggestion: (non blocking)** wrap these two new list items using `ftl_has_messages()`?

alexgibson reviewed (changes requested):
Updates look good! I'm not sure the new link on the download page is quite working as expected depending on operating system. But other than that, just a few small suggestions.

alexgibson commented:
Oh one other question: when is this all due to launch? Does any of it need to be behind a switch?

wen-2018 commented on media/img/products/vpn/download/platform-firefox-extension.svg:
Waiting for a better image

wen-2018 commented:
Hi @alexgibson I have updates my PR with the following changes:
- on the feature page, three screenshots were replaced with new ones, requested by stakeholder
- on the download page, I have implemented a new design showing to only Windows users (waiting for a better svg for the new section at the moment)
- for the feature switch, still waiting for a stakeholder response on whether we can launch immediately once we have approval for changes

alexgibson reviewed (commented):
Updates look great to me. Let me know when the last bits are in and I'll give it once last look before an r+

alexgibson reviewed (approved):
r+wc (unsure if the switch is needed, if not then r+)

slightlyoffbeat commented:
There is an image that needs to be updated. It has been updated on the [figma](https://www.figma.com/design/3Sg25N3kx6oSXHtYA2FXN3/VPN-Updates-Feb-2025?node-id=2-115&t=wuRzvFbxDO0fJbOq-1). 

It is the image for the new feature on the feature page.

I'll upload it here as well:

Section:

![image](https://github.com/user-attachments/assets/1619b7ae-c82e-490f-8514-3187a2c3ae1c)

Photo:


![vpnfirefox](https://github.com/user-attachments/assets/46bea20a-c9b2-4198-8e5b-87d35f2b61ab)

wen-2018 commented:
Waiting for a high quality image and confirmation that copy doc is the source of truth

wen-2018 commented:
Stakeholder approved changes.
